### DISCUSSION TITLE
核心已经更新，但显示UnblockNeteaseMusic 未在运行

### DISCUSSION THREAD

**Discussion by Vc-li** [UnblockNeteaseMusic/luci-app-unblockneteasemusic#130] [Q&A] [answered]
unm-debug
Launching luci-app-unblockneteasmusic Debugging Tool...

OpenWrt info:
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='SNAPSHOT'
DISTRIB_TARGET='x86/64'
DISTRIB_ARCH='x86_64'
DISTRIB_TAINTS='no-all'
DISTRIB_REVISION='R22.2.2'
DISTRIB_DESCRIPTION='Compiled on Mar.06,2022 @ OpenWrt '

uclient-fetch info:
Package: uclient-fetch
Version: 2021-05-14-6a6011df-1
Depends: libc, libuclient20201210
Provides: wget
Status: install user installed
Section: net
Architecture: x86_64
Size: 7941
Filename: uclient-fetch_2021-05-14-6a6011df-1_x86_64.ipk
Description: Tiny wget replacement using libuclient
Installed-Time: 1646446245

Package: libustream-wolfssl20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20211120, libwolfssl5.1.1.5fb91bea
Provides: libustream-wolfssl
Conflicts: libustream-openssl
Status: unknown ok not-installed
Section: libs
Architecture: x86_64
Size: 5064
Filename: libustream-wolfssl20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (wolfssl)

Package: libustream-openssl20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20210516, libopenssl1.1
Provides: libustream-openssl
Status: install user installed
Section: libs
Architecture: x86_64
Size: 5521
Filename: libustream-openssl20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (openssl)
Installed-Time: 1646446245

Package: libustream-mbedtls20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20211120, libmbedtls12
Provides: libustream-mbedtls
Conflicts: libustream-openssl, libustream-wolfssl
Status: unknown ok not-installed
Section: libs
Architecture: x86_64
Size: 5338
Filename: libustream-mbedtls20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (mbedtls)

Downloading 'https://api.github.com/repos/UnblockNeteaseMusic/server/commits?sha=enhanced&path=precompiled'
Connecting to 20.205.243.168:443
Writing to stdout
-                      122k --:--:-- ETA
Download completed (126479 bytes)
2f3a55c295f752fa26975cb02b67545fcdab22a5

Node.js info:
Package: node
Version: v14.18.3-1
Depends: libc, libstdcpp6, libopenssl1.1, zlib, libnghttp2-14, libuv1, libcares, libatomic1
Status: install ok installed
Section: lang
Architecture: x86_64
Size: 10508301
Filename: node_v14.18.3-1_x86_64.ipk
Description: Node.js® is a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node.js' package ecosystem, npm, is the largest ecosystem of open source libraries in the world. *** The following preparations must be made on the host side. *** 1. gcc 6.3 or higher is required. 2. To build a 32-bit target, gcc-multilib, g++-multilib are required. 3. Requires libatomic package. (If necessary, install the 32-bit library at the same time.) ex) sudo apt-get install gcc-multilib g++-multilib
Installed-Time: 1646844300

Node.js is placed at /usr/bin/node
Node.js version: Not Found

luci-app-unblockneteasmusic info:
Package: luci-app-unblockneteasemusic
Version: 2.12-2
Depends: libc, busybox, dnsmasq-full, ipset, jsonfilter, node, uclient-fetch
Status: install user installed
Architecture: all
Conffiles:
 /etc/config/unblockneteasemusic e9413f9ce065cfb7b875d1948c6d3258772ab8854b7ccf193df49c3c8b6cca20
Installed-Time: 1646844300

-rw-r--r--    1 root     root         288 Mar 10 01:08 /etc/config/unblockneteasemusic
-rwxr-xr-x    1 root     root        8.5K Jan  6 13:19 /etc/init.d/unblockneteasemusic

/usr/share/unblockneteasemusic:
drwxr-xr-x    2 root     root        3.4K Mar 10 01:08 core
-rw-r--r--    1 root     root          41 Mar 10 01:08 core_local_ver
-rwxr-xr-x    1 root     root        4.7K Jan  6 13:19 debugging.sh
-rwxr-xr-x    1 root     root         348 Jan  6 13:19 log_check.sh
-rwxr-xr-x    1 root     root        2.9K Jan  6 13:19 update.sh

config unblockneteasemusic 'config'
        option music_source 'default'
        option use_custom_cookie '0'
        option auto_update '1'
        option update_time '3'
        option enable '1'
        option local_vip '1'
        option enable_flac '1'
        option replace_music_source 'lower_than_320kbps'
        option advanced_mode '0'

UnblockNeteaseMusic Node.js info:
Git HEAD version: 2f3a55c295f752fa26975cb02b67545fcdab22a5
Core version: Not Found
-rw-r--r--    1 root     root      468.3K Mar 10 01:08 app.js
-rw-r--r--    1 root     root      411.8K Mar 10 01:08 bridge.js
-rw-r--r--    1 root     root         802 Mar 10 01:08 ca.crt
-rw-r--r--    1 root     root         847 Mar 10 01:08 server.crt
-rw-r--r--    1 root     root         359 Mar 10 01:08 server.key

Netease networking info:
< Location: http://m10.music.126.net/20220310013527/a60298f1b03799417f3029e216396ef5/ymusic/obj/w5zDlMODwrDDiGjCn8Ky/3057940231/3d32/93a3/4db1/0757f3a378b41e09ca2767c1abd47afa.mp3
{"timestamp":1646845948574,"status":404,"error":"Not Found","message":"No message available","path":"/httpdns/v2/d"}

Port status:
No instance found on port 5200.
No instance found on port 5201.

Running info:
PROCD running status: running
Thread is not found.

Firewall info:
Chain netease_cloud_music (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             0.0.0.0/8           
RETURN     all  --  anywhere             10.0.0.0/8          
RETURN     all  --  anywhere             127.0.0.0/8         
RETURN     all  --  anywhere             169.254.0.0/16      
RETURN     all  --  anywhere             172.16.0.0/12       
RETURN     all  --  anywhere             192.168.0.0/16      
RETURN     all  --  anywhere             base-address.mcast.net/4 
RETURN     all  --  anywhere             240.0.0.0/4         
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_http src tcp dpt:www redir ports 5200
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_https src tcp dpt:https redir ports 5201

Name: neteasemusic
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0xf8729731
Size in memory: 280
References: 1
Number of entries: 2
Members:
59.111.19.33
59.111.160.195

Name: acl_neteasemusic_http
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0xf44999f0
Size in memory: 200
References: 1
Number of entries: 0
Members:

Name: acl_neteasemusic_https
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0x41438689
Size in memory: 200
References: 1
Number of entries: 0
Members:

dhcp-option=252,http://192.168.66.1:5200/proxy.pac
ipset=/.music.163.com/neteasemusic
ipset=/interface.music.163.com/neteasemusic
ipset=/interface3.music.163.com/neteasemusic
ipset=/apm.music.163.com/neteasemusic
ipset=/apm3.music.163.com/neteasemusic
ipset=/clientlog.music.163.com/neteasemusic
ipset=/clientlog3.music.163.com/neteasemusic

Testing source replacing...
{"data":[{"id":33190502,"url":null,"br":0,"size":0,"md5":null,"code":404,"expi":1200,"type":null,"gain":0.0,"fee":0,"uf":null,"payed":0,"flag":257,"canExtend":false,"freeTrialInfo":null,"level":null,"encodeType":null,"freeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"listenType":null},"freeTimeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"type":0,"remainTime":0},"urlSource":0}],"code":200}
curl: (7) Failed to connect to 192.168.66.1 port 5200 after 0 ms: Connection refused

curl: (52) Empty reply from server

Log is available at:

https://transfer.sh/lHONNJ/unm-debugging-output.txt

**Comment by 1715173329** [comment, marked as answer]
你的 nodejs 文件有问题，无法被正确执行
手动输入指令 node，看看是不是缺了依赖。

**Reply by Vc-li** [reply]
应该是有错误，请问此错误如何解决呢，万分感谢🙏
root@OpenWrt:~# node
Error relocating /usr/bin/node: _ZNSt19_Sp_make_shared_tag5_S_eqERKSt9type_info: symbol not found
Error relocating /usr/bin/node: _ZNSt19basic_ostringstreamIcSt11char_traitsIcESaIcEEC1Ev: symbol not found
Error relocating /usr/bin/node: _ZSt28__throw_bad_array_new_lengthv: symbol not found
Error relocating /usr/bin/node: _ZNSt18basic_stringstreamIcSt11char_traitsIcESaIcEEC1Ev: symbol not found

**Reply by 1715173329** [reply]
make clean，重新编译
